### PR TITLE
memory: align buffers on 0x80 byte boundaries

### PR DESCRIPTION
**Pull request by lyakh**
One of typically allocated buffer sizes is 0x180 bytes. With buffers of 0x100 bytes this means, that 0x80 (128) bytes are wasted with each such allocation. Change block size to 0x80 bytes to avoid wasting memory.

### CODE REVIEW

File: src/platform/apollolake/include/platform/lib/memory.h [hunk 1]
```diff
@@ -292,7 +292,7 @@
```
Comment (dbaluta) [inline src/platform/apollolake/include/platform/lib/memory.h:292]: @lyakh what is the problem are we trying to solve. One very interesting things that I noticed on IMX and Cavs platforms is PLATFORM_DCACHE_ALIGN.

Runtime memory is always aligned and PLATFORM_DCACHE_ALIGN can be as much as 128 or 64 this means blocks with sizes less than 128 or 64 are rarely used on cAVS and IMX platforms.

This was actually a problem for iMX where we quickly run out of memory. On iMX we solved the problem by reducing PLATFORM_DCACHE_ALIGN to sizeof(void*). Maybe this is helpful for your problem.

But the main question here is WHY do we need this change now and how did you get to create such patch.

Comment (lyakh) [reply src/platform/apollolake/include/platform/lib/memory.h:292]: @dbaluta this patch isn't for arbitrary allocations, it's for DMA buffers. I added debug prints and I see 0x180 and 0x300 byte large DMA buffers being allocated. So, every time a 0x180 byte buffer is allocated, two 0x100 byte large chunks are occupied, which means 0x80 bytes are wasted. That's exactly the reason (I assume) why a non-intuitive size of 0x180 bytes was used before #2425

Comment (dbaluta) [reply src/platform/apollolake/include/platform/lib/memory.h:292]: @lyakh OK, I see. I'm OK with the patch. Out of curiosity is there a bug associated with this?

**Review by dbaluta — APPROVED** [review]

**Comment by plbossart** [comment]
Shouldn't the alignment be a property of the platform? Different platforms might have different cache lines or optimizations, why would 128 be good in all cases?

**Comment by lyakh** [comment]
> Shouldn't the alignment be a property of the platform? Different platforms might have different cache lines or optimizations, why would 128 be good in all cases?

@plbossart this is basically HDA alignment, that we discussed earlier AFAIU

File: src/platform/apollolake/include/platform/lib/memory.h [hunk 2]
```diff
@@ -292,7 +292,7 @@
 
 #define HEAP_BUFFER_SIZE	0x10000
 
-#define HEAP_BUFFER_BLOCK_SIZE	0x100
+#define HEAP_BUFFER_BLOCK_SIZE	0x80
```
Comment (lgirdwood) [inline src/platform/apollolake/include/platform/lib/memory.h:295]: I would also place a static assert here for cache line size (all platforms) and HDA block size (CAVS platforms). This way we will throw up a build error if someone tries to change this in the future to an incompatible block size.

**Comment by paulstelian97** [comment]
Is this still needed?

**Pull request closed without merging**